x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -939,11 +939,14 @@ public void testGettingTimeWithoutCalendar() throws Exception {
         });
         Long randomLongDate = randomNonNegativeLong();
         indexSimpleDocumentWithTrueValues(randomLongDate);
-        
+
         doWithQuery(SELECT_ALL_FIELDS, (results) -> {
             results.next();
 
-            java.sql.Time expectedTime = new java.sql.Time(randomLongDate % 86400000L);
+            ZoneId zoneId = getZoneFromOffset(randomLongDate);
+            java.sql.Time expectedTime = new java.sql.Time(
```

Comment:
Shouldn't this be extracted in a utils class?

Reply:
Didn't do it as it's used only once here, but I can extract it.